### PR TITLE
Add a sample showing partial page updates in Razor Pages

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @DamianEdwards

File: samples/Mvc.RazorPagePartial/Pages/Index.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        public string Message => DateTime.Now.ToString();
+
+        public IActionResult OnGet() => Page();
+
+        public IActionResult OnGetPartial()
+        {
+            return new PartialViewResult
+            {
+                ViewName = "_IndexPartial",
+                ViewData = new ViewDataDictionary<IndexModel>(this.ViewData, this),
```
Comment (DamianEdwards) [inline samples/Mvc.RazorPagePartial/Pages/Index.cshtml.cs]: So this is the magic incantation. Would be nice to have a built-in helper for this of course as this isn't particularly discoverable (passing in two `this` members, plus the current type as the generic arg, eeep).

File: samples/Mvc.RazorPagePartial/Pages/Index.cshtml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@page "{handler?}"
+@model IndexModel
+
+<div id="ajax-partial">
```
Comment (DamianEdwards) [inline samples/Mvc.RazorPagePartial/Pages/Index.cshtml]: Is this effectively identical to how it works in MVC of yore?

File: samples/Mvc.RazorPagePartial/Pages/Index.cshtml.cs (hunk 1 above)
Comment (pranavkm) [reply samples/Mvc.RazorPagePartial/Pages/Index.cshtml.cs]: Yup. Basically need https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs#L158-L209 on PageModel

File: samples/Mvc.RazorPagePartial/Pages/Index.cshtml (hunk 2 above)
Comment (pranavkm) [reply samples/Mvc.RazorPagePartial/Pages/Index.cshtml]: As far as I can tell, yes. There's obviously niceties around generating the link and form `Ajax.BeginForm` \ `Ajax.ActionLink`, but the remainder of it looks to be pretty consistent with Mvc 5.

**Pull request merged**